### PR TITLE
added metaparameters

### LINKED ISSUE #58: [9100_basics] Add information about `count`
Currently information about working with `count` in a resource is missing.

### PR DESCRIPTION
**Pull request by MrStrix**
fixes #58

### CODE REVIEW

File: terraform-workshop/9100_basics/modul_basics_9100.md [hunk 1]
```diff
@@ -253,6 +253,89 @@ Defines terraform configuration.
...
+}
+```
+
+Creates multiple instances of the resource
+
+## depends_on
+```hcl
+resource "azurerm_virtual_machine" "main" {
+  # forces creation of the network interface before the VM
+  depends_on = [ "${azurerm_network_interface.main}" ]
```
Comment (tongpu) [inline terraform-workshop/9100_basics/modul_basics_9100.md]: Using a dependency to a module might be more realistic as an example.

File: terraform-workshop/9100_basics/modul_basics_9100.md [hunk 2]
```diff
@@ -253,6 +253,89 @@ Defines terraform configuration.
...
+resource "azurerm_dns_a_record" "website" {
+  create_before_destroy = true
+}
+```
+
+Force creation of a new resource *before* the old resource is deleted. Useful for example for DNS records.
+
+## prevent_destroy
+```hcl
+resource "azurerm_virtual_machine" "main" {
```
Comment (hairmare) [inline terraform-workshop/9100_basics/modul_basics_9100.md]:
```suggestion
resource "azurerm_kubernetes_cluster" "main" {
```
Can has aks example pls?

File: terraform-workshop/9100_basics/modul_basics_9100.md (hunk 1 above)
Comment (hairmare) [reply terraform-workshop/9100_basics/modul_basics_9100.md]: I kinda like how this demonstrates forced recreation of the nic whereas recreating some modules might wreak more havoc. WDYT

File: terraform-workshop/9100_basics/modul_basics_9100.md [hunk 3]
```diff
@@ -253,6 +253,89 @@ Defines terraform configuration.
 
 ---
 
+## meta-parameters
+Can be applied to all resource definitions regardless of type
+
+* count (not applicable to modules)
+* depends_on
```
[the remark this answers is not in the document]
Comment (MrStrix) [reply terraform-workshop/9100_basics/modul_basics_9100.md]: added the same way as for count

File: terraform-workshop/9100_basics/modul_basics_9100.md [hunk 4]
```diff
@@ -253,6 +253,89 @@ Defines terraform configuration.
...
+* provider
+* lifecycle
+  * create_before_destroy
+  * prevent_destroy
```
[the remark this answers is not in the document]
Comment (MrStrix) [reply terraform-workshop/9100_basics/modul_basics_9100.md]: feel free to create one. for now I leave it in there and trust in our speakers to explain the details

File: terraform-workshop/9100_basics/modul_basics_9100.md [hunk 5]
```diff
@@ -253,6 +253,89 @@ Defines terraform configuration.
...
+```hcl
+resource "azurerm_virtual_machine" "main" {
+  count = 3 # creates 3 virtual machines
+  name = "VM-${count.index}"
+}
+```
+
+Creates multiple instances of the resource
+
+## depends_on
```
[the remark this answers is not in the document]
Comment (MrStrix) [reply terraform-workshop/9100_basics/modul_basics_9100.md]: same thing for `count`: https://github.com/hashicorp/terraform/issues/953
This is based on terraform 0.11 and future releases must be considered when released

File: terraform-workshop/9100_basics/modul_basics_9100.md [hunk 6]
```diff
@@ -253,6 +253,89 @@ Defines terraform configuration.
...
+  depends_on = [ "${azurerm_network_interface.main}" ]
+}
+```
+
+Creates a dependency when default dependency management fails
+
+## provider
+```hcl
+provider "azurerm" {
+  alias = "us"
```
[the remark this answers is not in the document]
Comment (MrStrix) [reply terraform-workshop/9100_basics/modul_basics_9100.md]: added version attributes for providers and created issue for semver section: #60

**Pull request merged**